### DISCUSSION TITLE
overwriting silence in patterns

### DISCUSSION THREAD

**Discussion by larkob** [tidalcycles/strudel#158] [General]
I'm still trying to understand how silence works and how it can be "undone" (no use case, just out of curiousity). This time, I've tried to fill out a part in a pattern that initially was empty, and even though at first the result seemed to be promising using `legato`, once I added another `struct` function to split the cycle up again, I didn't get the expected result. See here an example where I compare a pattern using silence + legato with a pattern that has legato applied to the third item via mini-notation syntax. Again, is this working as designed? And why? Is there maybe some better way for me to inspect the results of patterns rather than the map/showWhole approach?

```
"c2 d2 e2 ~".legato("1 1 2 0").struct("x x x x").firstCycle().map(h=>h.showWhole())`
// ['0/1 -> 1/4: c2', '1/4 -> 1/2: d2', '1/2 -> 3/4: e2']
```
```
"c2 d2 e2@2".struct("x x x x").firstCycle().map(h=>h.showWhole())
// ['0/1 -> 1/4: c2', '1/4 -> 1/2: d2', '1/2 -> 3/4: e2', '3/4 -> 1/1: e2']
```

**Comment by felixroos** [comment]
you can visualize using `.pianoroll`: https://strudel.tidalcycles.org?0KNPD8AmV-ms

**Comment by larkob** [comment]
That's good to get the timing, but it also doesn't give me the full information. The strange thing is that `"c2 d2 e2@2"` and `"c2 d2 e2 ~".legato("1 1 2 0")` seem to produce exactly the same pattern for all that I can see with `show`, `drawLine` or `pianoroll`. So, how come the `struct` function produces a different result with them, as if there was some hidden information that Strudel knows about but doesn't reveal?
